### PR TITLE
auto is a valid option for vacuum index_cleanup.

### LINKED ISSUE #6032: Vacuum index_cleanup auto Error
```sql
VACUUM (INDEX_CLEANUP auto);
ERROR:  index_cleanup requires a Boolean value
```

### PR DESCRIPTION
**Pull request by aykut-bozkurt**
DESCRIPTION: Fixes #6032.

### CODE REVIEW

File: src/backend/distributed/commands/vacuum.c [hunk 1]
```diff
@@ -541,6 +564,14 @@ VacuumStmtParams(VacuumStmt *vacstmt)
 		#endif
 		else if (strcmp(opt->defname, "index_cleanup") == 0)
 		{
+			#if PG_VERSION_NUM >= PG_VERSION_14
+			if (opt->arg && strcmp(defGetString(opt), "auto") == 0)
```
Comment (onderkalaci) [inline src/backend/distributed/commands/vacuum.c:568]: this part seems a little hard to follow. Can't we decide `params.index_cleanup` in a single `if` or `case` statement? Now that the logic seems scattered

File: src/test/regress/expected/multi_utilities.out [hunk 2]
```diff
@@ -404,21 +404,6 @@ SELECT pg_size_pretty( pg_total_relation_size('local_vacuum_table') );
  21 MB
 (1 row)
 
------------------ PROCESS_TOAST is only available for pg14
```
Comment (onderkalaci) [inline src/test/regress/expected/multi_utilities.out:407]: why do we remove these tests?

**Review by onderkalaci — COMMENTED** [review]
minor comments, I'd like to have a look & do final testing before approving. Thanks!

File: src/test/regress/expected/multi_utilities.out (hunk 2 above)
Comment (aykut-bozkurt) [reply src/test/regress/expected/multi_utilities.out:407]: Process_toast tests are already in pg14.sql file. It should not be in multi_utilities.sql because auto option for index_cleanup is only valid for pg14.

File: src/backend/distributed/commands/vacuum.c (hunk 1 above)
Comment (onderkalaci) [reply src/backend/distributed/commands/vacuum.c:568]: First, we should follow the same logic as PG: https://github.com/postgres/postgres/blob/07d683b54af854098cc559d4b8640905f9efa0ea/src/backend/commands/vacuum.c#L151-L166


Such that `if (!opt->arg)` is also interpreted as `auto` or `auto` should be checked case insensitive via `pg_strcasecmp`. Can you also add a tests for these case? 

In terms of code, can't we do the following where we make this difference a lot more explicit to the user:
```C
    
		else if (strcmp(opt->defname, "index_cleanup") == 0)
		{
#if PG_VERSION_NUM >= PG_VERSION_14

			/* Interpret no string as the default, which is 'auto' */
			if (!opt->arg)
				params.index_cleanup = VACOPTVALUE_AUTO;
			else
			{
				char	   *sval = defGetString(opt);

				/* Try matching on 'auto' string, or fall back on boolean */
				if (pg_strcasecmp(sval, "auto") == 0)
					params.index_cleanup = VACOPTVALUE_AUTO;
				else
					params.index_cleanup = get_vacoptval_from_boolean(opt);
			}
		}
#else    
... /* old code */
#endif  
```

We can omit `get_vacoptval_from_boolean` function for this PR, but would be good clean-up for the next ones

File: src/test/regress/expected/multi_utilities.out (hunk 2 above)
Comment (onderkalaci) [reply src/test/regress/expected/multi_utilities.out:407]: But these tests are on a local table and potentially testing the behavior on the local tables. I don't think we should remove these. In other words, I'm not convinced that we should remove if you don't have a strong opinion on this.

If you really want to remove these, it is better to open a separate PR and describe why you want to remove so that we can focus on those.

Comment (aykut-bozkurt) [reply src/test/regress/expected/multi_utilities.out:407]: Those tests are added by myself when I work on #5946. They never run before because they were commented out. When I remove comments, they fail for postgres build 13 because there is no support for process_toast on pg13. Thus, we need to run process toast tests only for postgres version 14. That is why I moved those process toast tests to pg14.sql file.

File: src/backend/distributed/commands/vacuum.c [hunk 3]
```diff
@@ -380,10 +380,33 @@ DeparseVacuumStmtPrefix(CitusVacuumParams vacuumParams)
...
 	{
-		appendStringInfoString(vacuumPrefix,
-							   vacuumParams.index_cleanup == VACOPTVALUE_ENABLED ?
-							   "INDEX_CLEANUP," : "INDEX_CLEANUP false,"
-							   );
+		switch (vacuumParams.index_cleanup)
+		{
+			case VACOPTVALUE_ENABLED:
+			{
+				appendStringInfoString(vacuumPrefix, "INDEX_CLEANUP,");
```
Comment (onderkalaci) [inline src/backend/distributed/commands/vacuum.c:387]: I'm a little confused by this.

Why don't we send `INDEX_CLEANUP on` when `VACOPTVALUE_ENABLED`? If we don't add anything, PG on the workers would qualify that as `auto`, which is wrong?

File: src/test/regress/expected/multi_utilities.out (hunk 2 above)
Comment (onderkalaci) [reply src/test/regress/expected/multi_utilities.out:407]: My point is that pg14.sql file does NOT have as detailed tests as this one, so why not we move these tests there?

I'm fine if you share the exact tests that are on pgg14.sql that completely covers here. Otherwise, I'd suggest we move these tests to PG 14.sql file and uncomment them

**Review by onderkalaci — APPROVED** [review]
Thanks for the fix and the test coverage. 

(We can potentially backport this if ever any user brings up)
